diff --git a/TransformOperationExecutor.js b/TransformOperationExecutor.js
index dec36c7e5b5c1855351279c9581d4065d3ab2f0f..d6bd09aa170401ea66596cbe1db23056002fc01d 100644
--- a/TransformOperationExecutor.js
+++ b/TransformOperationExecutor.js
@@ -38,6 +38,11 @@ class TransformOperationExecutor {
     // Public Methods
     // -------------------------------------------------------------------------
     transform(source, value, targetType, arrayType, isMap, level = 0) {
+        // Check if the value is a promise, and return if it is instead of failing
+        // See: https://github.com/typestack/class-transformer/issues/261
+        if (value && typeof value === 'object' && typeof value.then === 'function'){
+            return value;
+        }
         if (Array.isArray(value) || value instanceof Set) {
             const newValue = arrayType && this.transformationType === enums_1.TransformationType.PLAIN_TO_CLASS
                 ? instantiateArrayType(arrayType)
